edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/EdgeHubScopeDevice.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Http.Controllers
+{
+    using System.Collections.Generic;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Shared;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+
+    public class EdgeHubScopeDevice
+    {
+        public EdgeHubScopeDevice(
+            string deviceId,
+            string generationId,
+            DeviceStatus status,
+            AuthenticationMechanism authentication,
+            DeviceCapabilities capabilities,
+            string deviceScope,
+            IEnumerable<string> parentScopes)
+        {
+            this.Id = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
+            this.GenerationId = Preconditions.CheckNonWhiteSpace(generationId, nameof(generationId));
+            this.Status = Preconditions.CheckNotNull(status);
+            this.Authentication = Preconditions.CheckNotNull(authentication);
+            this.Capabilities = Preconditions.CheckNotNull(capabilities);
+            this.Scope = deviceScope;
+            this.ParentScopes = parentScopes;
+        }
+
+        [JsonProperty(PropertyName = "deviceId")]
+        public string Id { get; set; }
+
+        [JsonProperty(PropertyName = "generationId")]
+        public string GenerationId { get; set; }
+
+        [JsonConverter(typeof(StringEnumConverter))]
+        [JsonProperty(PropertyName = "status")]
+        public DeviceStatus Status { get; set; }
+
+        [JsonProperty(PropertyName = "authentication")]
+        public AuthenticationMechanism Authentication { get; set; }
+
+        [JsonProperty(PropertyName = "capabilities", NullValueHandling = NullValueHandling.Ignore)]
```

Comment:
nit: don't need NullValueHandling as it is not allowed to be null in constructor.